test/gtest/ucp/test_ucp_am.cc

```diff
@@ -799,6 +799,188 @@ UCS_TEST_P(test_ucp_am_nbx, rx_am_mpools,
 UCP_INSTANTIATE_TEST_CASE(test_ucp_am_nbx)
 
 
+class test_ucp_am_nbx_send_copy_header : public test_ucp_am_nbx {
```

Comment:
why do we need separate tests? Can we add another variant (with new flag) to some of the test classes?

Reply:
Hi. Maybe I can merge it with test_ucp_am_nbx that it's heavily rely on (Like many other tests in this file...) for example, 
The question is if this test not deserve a separate class like other tests? 
It's doing some unique things for example:
1. It's sending more than one message to fill the send queue.
2. The validation of the messages is different (with counters).
3. There are two different flows, eager and rndv. (I can maybe merge the two flows into one function using flags, I like it the way it is, but combine them into one function will increase code reuse).

You think it's better to remove this new class and to add the changes into some existent one?

Reply:
maybe adding one test to the existing infra which ensures pending flow is affected would be shorter?